src/api/app/controllers/source_controller.rb

```diff
@@ -521,7 +521,7 @@ def check_and_remove_repositories!(repositories, full_remove, force = false)
     if !force && error[:error]
       raise RepoDependency, error[:error]
     else
-      error = Project.remove_repositories(repositories, full_remove)
+      error = Project.remove_repositories(repositories, full_remove, true)
```

Comment:
or even easier ... we never call that method with writing to backend ... just drop it.